AFQ/api/participant.py

```diff
@@ -0,0 +1,130 @@
+import nibabel as nib
+
+from AFQ.definitions.mapping import SlrMap
+from AFQ.api.utils import wf_sections, add_method_descriptors
+
+from AFQ.tasks.data import get_data_plan
+from AFQ.tasks.mapping import get_mapping_plan
+from AFQ.tasks.tractography import get_tractography_plan
+from AFQ.tasks.segmentation import get_segmentation_plan
+from AFQ.tasks.viz import get_viz_plan
+
+
+__all__ = ["ParticipantAFQ"]
+
+
+@add_method_descriptors
+class ParticipantAFQ(object):
+    def __init__(self,
+                 dwi_data_file,
+                 bval_file, bvec_file,
+                 output_dir,
+                 bids_info=None,
+                 **kwargs):
+        """
+        Initialize a ParticipantAFQ object from a BIDS dataset.
+        Some special notes on parameters:
+        In tracking_params, parameters with the suffix mask which are also
+        a mask from AFQ.definitions.mask will be handled automatically by
+        the api.
+
+        Parameters
+        ----------
+        dwi_data_file : str
+            Path to DWI data file.
+        bval_file : str
+            Path to bval file.
+        bvec_file : str
+            Path to bvec file.
+        output_dir : str
+            Path to output directory.
+        bids_info : dict or None, optional
+            This is used by GroupAFQ to provide information about
+            the BIDS layout to each particpant. It is reccomended
+            that you leave this parameter as None,
+            and instead pass in the paths
+            to the files you want to use directly.
+        kwargs : additional optional parameters
+            You can set additional parameters for any step
+            of the process. See `The pyAFQ API optional arguments`
+            in the `usage` section of pyAFQ's documentation for more
+            details.
+            For example, to set the sh_order for csd to 4, do:
```

Comment:
You can add an "Examples" section to the docstring to show some examples of setting kwargs.